templates/cluster-template-vpc-clusterclass.yaml

```diff
@@ -0,0 +1,422 @@
+apiVersion: cluster.x-k8s.io/v1beta1
+kind: Cluster
+metadata:
+  labels:
+    ccm: external
+    cluster.x-k8s.io/cluster-name: ${CLUSTER_NAME}
+  name: ${CLUSTER_NAME}
+  namespace: ${NAMESPACE}
+spec:
+  clusterNetwork:
+    pods:
+      cidrBlocks:
+      - ${POD_CIDR:="192.168.0.0/16"}
+    serviceDomain: ${SERVICE_DOMAIN:="cluster.local"}
+    services:
+      cidrBlocks:
+      - ${SERVICE_CIDR:="10.128.0.0/12"}
+  topology:
+    class: ${IBMVPC_CLUSTER_CLASS_NAME}
+    controlPlane:
+      replicas: ${CONTROL_PLANE_MACHINE_COUNT}
+    version: ${KUBERNETES_VERSION}
+    workers:
+      machineDeployments:
+        - class: default-worker
+          name: md-0
+          replicas: 2
+---
+apiVersion: cluster.x-k8s.io/v1beta1
+kind: ClusterClass
+metadata:
+  name: ${IBMVPC_CLUSTER_CLASS_NAME:="vpc-cc"}
+spec:
+  controlPlane:
+    machineInfrastructure:
+      ref:
+        apiVersion: infrastructure.cluster.x-k8s.io/v1beta2
+        kind: IBMVPCMachineTemplate
+        name: ${IBMVPC_CLUSTER_CLASS_NAME}-control-plane-machinetemplate
+        namespace: ${NAMESPACE}
+    ref:
+      apiVersion: controlplane.cluster.x-k8s.io/v1beta1
+      kind: KubeadmControlPlaneTemplate
+      name: ${IBMVPC_CLUSTER_CLASS_NAME}-control-plane
+      namespace: ${NAMESPACE}
+  infrastructure:
+    ref:
+      apiVersion: infrastructure.cluster.x-k8s.io/v1beta2
+      kind: IBMVPCClusterTemplate
+      name: ${IBMVPC_CLUSTER_CLASS_NAME}-cluster-template
+      namespace: ${NAMESPACE}
+  workers:
+    machineDeployments:
+      - class: default-worker
```

Comment:
For testing lets create another class of workers may be using another instance profile and try using it cluster.

Reply:
Noted, will test this flow.

Reply:
This has been verified:
<img width="1170" alt="Pasted Graphic 7" src="https://github.com/kubernetes-sigs/cluster-api-provider-ibmcloud/assets/110517346/ad52d867-75fb-44f4-a7fa-7fbcba235d06">
All nodes ready
<img width="851" alt="Pasted Graphic 8" src="https://github.com/kubernetes-sigs/cluster-api-provider-ibmcloud/assets/110517346/56b5fd86-17c6-4e88-af57-9c18e8191a0e">

Reply:
Awesome, Thanks for confirming, Lets add necessary documentation and close this